terrawrap/utils/config.py

```diff
@@ -86,6 +88,127 @@ def parse_wrapper_configs(wrapper_config_files: List[str]) -> WrapperConfig:
+    """
+    wrapper_file = os.path.join(directory, ".tf_wrapper")
+    with open(wrapper_file) as wrapper_file:
+        wrapper_config = yaml.safe_load(wrapper_file)
+        return "depends_on" in wrapper_config
+
+
+def parse_dependencies(directory: str) -> List[str]:
+    """
+    Function for parsing the Terraform wrapper config file for dependencies.
+    :param directory: A directory containing a tf_wrapper.
+    :return: dependencies - A list of dependencies for the given directory.
+    """
+    dependencies = []
+
+    for file_name in os.listdir(directory):
+        if file_name.endswith('.tf_wrapper'):
+            file_name = os.path.join(directory, file_name)
+            wrapper_config_file = file_name
+            with open(wrapper_config_file) as wrapper_config_file:
+                wrapper_config = yaml.safe_load(wrapper_config_file)
+
+            try:
+                wrapper_dependencies = wrapper_config.get('depends_on')
+                if wrapper_dependencies:
+                    for path in wrapper_dependencies:
+                        path = get_absolute_path(path)
+                        dependencies.append(path)
+                    return dependencies
+            except TypeError:
+                return dependencies
+    return dependencies
+
+
+def single_config_dependency_grapher(config_dir: str, graph: networkx.DiGraph, visited: List[str]):
+    """
+    Given a directory, recursively finds all other directories it depends on and builds a graph.
+    :param config_dir: The config directory to obtain a dependency graph for
+    :param graph: The graph to add dependency info to. Empty at first, reused in recursion.
+    :param visited: A list of visited nodes. Empty at first, reused in recursion
+    """
+    if config_dir in visited:
+        return
+    visited.append(config_dir)
+    if is_config(config_dir):
+        graph.add_node(config_dir)
+
+    tf_dependencies = parse_dependencies(config_dir)
+    for dependency in tf_dependencies:
+        graph.add_node(dependency)
+        if config_dir in graph:
+            graph.add_edge(dependency, config_dir)
+
+    wrappers = find_wrapper_config_files(config_dir)
+    wrappers.reverse()  # we want the closest wrapper file that gives inherited dependencies
+    for wrapper in wrappers:
+        wrapper_dir = os.path.dirname(wrapper)
+        if wrapper_dir == config_dir:
+            continue
+        if has_depends_on(wrapper_dir):
+            closest_inheritance = wrapper_dir
+            inherited_dependencies = parse_dependencies(closest_inheritance)
+            added = False
+            for dependency in inherited_dependencies:
+                if dependency == config_dir:
+                    continue
+                added = True
+                graph.add_node(dependency)
+                if config_dir in graph:
+                    graph.add_edge(dependency, config_dir)
+            if added:
```

Comment:
Don't we want this to be inside the for loop?

Reply:
Also wondering if you actually need the check to see if `wrapper_dir == config_dir` and `depencency == config_dir` since you first check if the config_dir has been visited.

Reply:
This has to be outside to make sure that we are actually adding something new, i ran into an edge case where if we have a level above that has deps we already encountered, it would break out of the loop eventhough it needs to check further up and it wasnt visited yet

Reply:
the dependecy could be the config directory and not visited